src/System.Memory/tests/Span/Sort.cs

```diff
@@ -0,0 +1,1606 @@
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_UInt32_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_UInt32_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_Int64_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_Int64_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_UInt64_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_UInt64_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_Single_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_Single_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_Double_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_Double_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_Boolean_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_Boolean_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_Char_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_Char_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_String_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_String_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_ComparableStructInt32_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_ComparableStructInt32_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_ComparableClassInt32_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_ComparableClassInt32_Int32(sortCases);
+        }
+
+        [Theory]
+        [Trait(SortTrait, SortTraitValue)]
+        [MemberData(nameof(s_fastSortTests))]
+        public static void Sort_KeysValues_BogusComparable_Int32(ISortCases sortCases)
+        {
+            Test_KeysValues_BogusComparable_Int32(sortCases);
+        }
+#if OUTER_LOOP
```

Comment:
This file is quite large. Consider splitting it up into partial class/files.

Maybe separate all outerloop tests?

Reply:
splitting out helper code instead, and fillers in two separate files